### PR TITLE
rough wip

### PR DESCRIPTION
**Pull request by vinitkumar**

### CODE REVIEW

File: divio_cli/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '3.3.0'
+__version__ = '3.3.2'
```
Comment (Chive) [inline divio_cli/__init__.py]: please don't bump the version yet - we'll do that in a separate commit.

File: divio_cli/cli.py [hunk 2]
```diff
@@ -429,6 +429,24 @@ def push_db(obj, stage, noinput):
...
+    test (default) or live
+    """
+    website_id = obj.project["id"]
+    if not noinput:
+        click.secho(messages.PUSH_DB_WARNING.format(stage=stage), fg='red')
+        if not click.confirm('\nAre you sure you want to continue?'):
+            return
+    localdev.push_local_db(obj.client, stage, dump_filename, website_id)
+
+
```
Comment (Chive) [inline divio_cli/cli.py]: I'd rather have this entire command as a flag to the existing function ``push_db``. Something like "if supplied a ``--file`` or ``--dump`` argument, the file is used instead of the one in the local db container". "Push local db" is a bit confusing since both the sql dump and the locally running database are "local".

File: divio_cli/localdev/main.py [hunk 3]
```diff
@@ -506,6 +506,34 @@ def dump_database(dump_filename, archive_filename=None):
     )
 
 
+def compress_db(dump_filename, archive_filename=None, archive_wd=None):
+
+    if not archive_filename:
+        # archive filename not specified
+        # return path to uncompressed dump
+        return os.path.join(archive_wd, dump_filename)
```
Comment (Chive) [inline divio_cli/localdev/main.py]: why?

File: divio_cli/localdev/main.py [hunk 4]
```diff
@@ -571,6 +599,62 @@ def push_db(client, stage):
     click.echo(' [{}s]'.format(int(time() - start_time)))
 
 
+def push_local_db(client, stage, dump_filename, website_id):
+    # TODO: This is not required, try and get the id from the passed
+    # params and use that to upload it ot client.
+    if not os.path.exists(dump_filename):
+        click.secho("The dumpfile {} doesn't seem to exists in the current directory".format(dump_filename))
```
Comment (Chive) [inline divio_cli/localdev/main.py]: doesn't seem to **exist**

File: divio_cli/localdev/main.py [hunk 5]
```diff
@@ -571,6 +599,62 @@ def push_db(client, stage):
...
+        dump_filename=dump_filename,
+        archive_filename=archive_filename,
+        archive_wd=archive_wd
+    )
+
+    click.secho(' ---> Uploading', nl=False)
+    start_upload = time()
+    assert client is not None
+    assert website_id is not None
+    assert archive_path is not None
```
Comment (Chive) [inline divio_cli/localdev/main.py]: not a huge fan of asserts since they don't provide any useful feedback to the user if they fail. furthermore they can be disabled. I'd rather have a meaningful error message here.

Comment (vinitkumar) [inline divio_cli/localdev/main.py]: @Chive Sorry for the confusion, created a fresh PR to the upstream. You are looking at the outdated one.  https://github.com/divio/divio-cli/pull/196 is the correct one.

File: divio_cli/cli.py (hunk 2 above)
Comment (vinitkumar) [inline divio_cli/cli.py]: @Chive Noted. Will do this change.

**Comment by Chive** [comment]
@vinitkumar No worries :) I've added my comments on the upstream version as well https://github.com/divio/divio-cli/pull/196
